### PR TITLE
Add entities for map-related features

### PR DESCRIPTION
**Pull request by 6167656e74323431**
As the name implies.

Note this requires #18 to be merged first.

### CODE REVIEW

**Review by siddharthgowda — APPROVED** [review]

**Review by siddharthgowda — APPROVED** [review]

**Comment by 6167656e74323431** [comment]
This is a certified [ohshitgit](https://ohshitgit.com/) moment.

**Comment by hoodiexxx** [comment]
Well organized, might have potential conflicting in Doorway.java

File: src/main/java/com/mg105/entities/Doorway.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Create a doorway with its position in the room and the next room.
+     *
+     * @param position the in-room position of this doorway.
+     * @param nextRoom the room that should be travelled to if this doorway is entered.
+     */
+    public Doorway(RoomPosition position, Room nextRoom) {
+        this.position = position;
+        this.nextRoom = nextRoom;
+    }
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/entities/Doorway.java:16]: nice way to access the room and next-room, but variable type for these instance variables should depends on the way we store the rooms.

**Review by hoodiexxx — APPROVED** [review]
well done. one thing I believe is we don't need 2D array for storing these rooms.

**Review by siddharthgowda — APPROVED** [review]

**Review by zhaoan12 — APPROVED** [review]

**Pull request merged**